core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -284,33 +290,57 @@ func (i *pluginOracleCreator) createReadersAndWriters(
 
 	contractReaders := make(map[cciptypes.ChainSelector]types.ContractReader)
 	chainWriters := make(map[cciptypes.ChainSelector]types.ChainWriter)
-	for _, chain := range i.chains.Slice() {
-		chainSelector, err1 := i.getChainSelector(chain.ID().Uint64())
+	for relayID, relayer := range i.relayers {
+		chainID, err1 := decimal.NewFromString(relayID.ChainID)
 		if err1 != nil {
-			return nil, nil, err1
+			return nil, nil, fmt.Errorf("error parsing chain ID, expected big int: %s %w", relayID.ChainID, err1)
+		}
+
+		chainSelector, err1 := i.getChainSelector(chainID.BigInt().Uint64())
+		if err1 != nil {
+			return nil, nil, fmt.Errorf("failed to get chain selector from chain ID %s: %w", chainID.String(), err1)
+		}
+
+		chainReaderConfig, err1 := getChainReaderConfig(chainID.BigInt().Uint64(), destChainID, homeChainID, ofc, chainSelector)
+		if err1 != nil {
+			return nil, nil, fmt.Errorf("failed to get chain reader config: %w", err1)
 		}
 
-		chainReaderConfig := getChainReaderConfig(chain.ID().Uint64(), destChainID, homeChainID, ofc, chainSelector)
-		cr, err1 := createChainReader(i.lggr, chain, chainReaderConfig, pluginType)
+		// TODO: context.
```

Comment:
What do you mean with this comment? Should the context be passed at a higher level to this method?

Reply:
Yeah, we should figure out where to pipe the context from.